numba/targets/builtins.py

```diff
@@ -511,6 +511,24 @@ def optional_is_not_none(context, builder, sig, args):
                   types.none, types.Optional)(optional_is_none))
 
 
+@builtin_attr
+@impl_attribute_generic(types.Optional)
+def optional_getattr(context, builder, typ, value, attr):
```

Comment:
`numba.targets.optional` only contains helpers for optional types while the rest of the implementation of optional are in `numba.targetsbuiltins` just before these lines.  We can move them in a separate PR.